test_util/cluster_api.py

```diff
@@ -338,6 +340,11 @@ def get_test_app_in_docker(self, ip_per_container):
         return app, test_uuid
 
     def deploy_test_app_and_check(self, app, test_uuid):
+        marathon_user = self.mesos_user
+        # if using docker containerizer, reported user will be docker user
+        if app['container'] and app['container']['type'] == 'DOCKER':
+            marathon_user = 'root'
+        marathon_user = self.default_os_user
```

Comment:
Seems like you should have an `else` here, rather than always overriding the `marathon_user` with the `default_os_user`.